### PR TITLE
Add registration login logout

### PR DESCRIPTION
**Pull request by atreidas1**

### CODE REVIEW

File: items/resources.py [hunk 1]
```diff
@@ -179,25 +190,25 @@ def prepend_urls(self):
...
-        user = User.objects.create_user(username=username, password=password, email=email)
-      except IntegrityError:
-        return HttpResponseServerError('There is user with this name')
-      response = self.create_response(request, Unpacking.unpack_user(user))
-      return HttpResponse(response)
-    else:
-      return HttpResponseServerError('Forbidden method')
+    self.method_check(request, allowed=['post'])
+    data = self.deserialize(request, request.body,
+    format=request.META.get('CONTENT_TYPE', 'application/json'))
```
Comment (ghost) [inline items/resources.py]: it looks like `format` is never used

File: items/test_models.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+from django.test import TestCase
+from .models import Item, Group
+from django.contrib.auth.models import User
+from django.utils import timezone
+from django.db import IntegrityError
+
+
+class ItemModelTest(TestCase):
```
Comment (ghost) [inline items/test_models.py:8]: Name of the class doesn't reflect content. You can break tests by classes

File: items/resources.py [hunk 3]
```diff
@@ -145,28 +145,39 @@ def prepend_urls(self):
...
+      return self.create_response(request,{}, HttpUnauthorized)
 
   def get_user(self, request, **kwargs):
     user = request.user
-    response = self.create_response(request, Unpacking.unpack_user(user))
-    return HttpResponse(response)
+    if user.is_anonymous():
+      return self.create_response(request, {}, HttpUnauthorized)
+    else:
+      return self.create_response(request,{'user' : Unpacking.unpack_user(user)})
```
Comment (ghost) [inline items/resources.py]: if you look into pep8 or pyflakes you'll see this code is not passed. It's kind of lintering for python

File: items/resources.py [hunk 4]
```diff
@@ -145,28 +145,39 @@ def prepend_urls(self):
...
   def login(self, request, **kwargs):
-    if request.method == 'POST':
-      data = json.loads(request.body)
-      username = data['username']
-      password = data['password']
+    self.method_check(request, allowed=['post'])
+    current_user = request.user
+    if current_user.is_anonymous():
+      data = self.deserialize(request, request.body,
+      format=request.META.get('CONTENT_TYPE', 'application/json'))
```
Comment (NikitaSedyx) [inline items/resources.py]: the same with `format`

File: items/resources.py [hunk 5]
```diff
@@ -179,25 +190,25 @@ def prepend_urls(self):
...
+    password = data.get('password')
+    repeatPassword = data.get('confirmPassword')
+    try:
+      user=User.objects.get(username=username)
+      return self.create_response(request, { 'success' : False })
+    except ObjectDoesNotExist:
+      if password == repeatPassword:
+        email = data.get('email')
+        user = User.objects.create_user(username, email, password)
+        user.save()
```
Comment (NikitaSedyx) [inline items/resources.py]: it's necessary here

File: items/resources.py [hunk 6]
```diff
@@ -179,25 +190,25 @@ def prepend_urls(self):
...
+        user = auth.authenticate(username=username, password=password)
+        auth.login(request,user)
```
Comment (NikitaSedyx) [inline items/resources.py]: spaces

File: items/resources.py [hunk 7]
```diff
@@ -179,25 +189,23 @@ def prepend_urls(self):
...
-    else:
-      return HttpResponseServerError('Forbidden method')
+    self.method_check(request, allowed=['post'])
+    data = self.deserialize(request, request.body, format=request.META.get('CONTENT_TYPE', 'application/json'))
+    username = data.get('username')
+    password = data.get('password')
+    repeatPassword = data.get('confirmPassword')
+    try:
+      user=User.objects.get(username=username)
+      return self.create_response(request, { 'success': False })
```
Comment (NikitaSedyx) [inline items/resources.py]: don't use spaces in {}, pls

File: items/resources.py [hunk 8]
```diff
@@ -145,28 +145,38 @@ def prepend_urls(self):
...
+      username = data.get('username')
+      password = data.get('password')
       user = auth.authenticate(username=username, password=password)
       if user is not None:
-        auth.login(request, user)
-        response = self.create_response(request, Unpacking.unpack_user(user))
-        return HttpResponse(response)
+        if user.is_active:
+          auth.login(request,user)
+          return self.create_response(request, {'user' : Unpacking.unpack_user(user)})
```
Comment (NikitaSedyx) [inline items/resources.py]: don't use space before `:`

File: items/test_models.py (hunk 2 above)
Comment (atreidas1) [inline items/test_models.py:8]: This tests writed by Nikita.This file appeared after squash.It already in master branch.

Comment (ghost) [inline items/test_models.py:8]: this file in you PR. PR - is what exactly you are going to merge to master. It's a diff between your branch and master. So it's your code or you failed with merging code from master to your branch

Comment (alekseyC64) [inline items/test_models.py]: Also the squash of commits in your branch is better to do after PR is reviewed

Comment (atreidas1) [inline items/test_models.py:8]: probably my failed with mergin

**Pull request merged**